src/main/java/duke/commands/DukeCommandFind.java

```diff
@@ -0,0 +1,149 @@
+package duke.commands;
+
+import duke.exceptions.DukeExceptionIllegalArgument;
+import duke.parser.Parser;
+import duke.storage.FileLoader;
+import duke.tasks.DateTask;
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+import duke.ui.Ui;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Find command.
+ *
+ * Filters tasks according to supplied date range and displays list to user.
+ */
+public class DukeCommandFind extends DukeCommand {
+
+    protected Optional<LocalDateTime> from;
+    protected Optional<LocalDateTime> to;
+    protected String search;
+
+    public DukeCommandFind(String input) throws DukeExceptionIllegalArgument {
+        // Process any keywords first, assumed stripped
+        search = "";
+        if (!input.startsWith("/from ")&&(!input.startsWith("/to "))) {
+            int searchIndex = input.length();
+            int fromIndex = input.indexOf(" /from");
+            int toIndex = input.indexOf(" /to");
+            if (fromIndex != -1 && searchIndex > fromIndex) {
+                searchIndex = fromIndex;
+            }
+            if (toIndex != -1 && searchIndex > toIndex) {
+                searchIndex = toIndex;
+            }
+
+            search = input.substring(0, searchIndex);
+            input = input.substring(searchIndex).strip();
+            input = input.strip();
+        }
+
+        // Date parsing
+        from = Optional.empty();
+        to = Optional.empty();
+        if (input.startsWith("/from ")) {
+            input = input.substring(6);
+            if (input.contains(" /to ")) {
+                String[] dates = input.split(" /to ");
+                from = Optional.of(Parser.parseDate(dates[0].strip()));
+                to = Optional.of(Parser.parseDate(dates[1].strip()));
+            } else {
+                from = Optional.of(Parser.parseDate(input));
+            }
+        } else if (input.startsWith("/to ")) {
+            input = input.substring(4);
+            if (input.contains(" /from ")) {
+                String[] dates = input.split(" /from ");
+                to = Optional.of(Parser.parseDate(dates[0].strip()));
+                from = Optional.of(Parser.parseDate(dates[1].strip()));
+            } else {
+                to = Optional.of(Parser.parseDate(input));
+            }
+        }
+        if (from.isPresent() && to.isPresent() && from.get().isAfter(to.get())) {
+            throw new DukeExceptionIllegalArgument(
+                    "'/from' date must be before '/to' date.");
+        }
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, FileLoader loader) throws DukeExceptionIllegalArgument{
+        boolean datedSearch = (from.isPresent() || to.isPresent());
```

Comment:
searchDate may be a better name

Reply:
or even better, `toSearchDate`. Good reco thanks!